aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -92,18 +97,111 @@ impl SessionId {
 
 pub struct SessionExt<'r, 'l, S> {
     inner: Session<'r, 'l, S>,
+    remote: MoveResolverWithVMMetadata<'r, 'l, S>,
 }
 
 impl<'r, 'l, S> SessionExt<'r, 'l, S>
 where
     S: MoveResolverExt,
 {
-    pub fn new(inner: Session<'r, 'l, S>) -> Self {
-        Self { inner }
+    pub fn new(inner: Session<'r, 'l, S>, move_vm: &'l MoveVM, remote: &'r S) -> Self {
+        Self {
+            inner,
+            remote: MoveResolverWithVMMetadata::new(remote, move_vm),
+        }
     }
 
     pub fn finish(self) -> VMResult<SessionOutput> {
         let (change_set, events, mut extensions) = self.inner.finish_with_extensions()?;
+
+        let common_error = PartialVMError::new(StatusCode::UNREACHABLE).finish(Location::Undefined);
+        let mut change_set_grouped = MoveChangeSet::new();
+        for (addr, account_changeset) in change_set.into_inner() {
+            let mut resource_groups: BTreeMap<StructTag, AccountChangeSet> = BTreeMap::new();
+            let mut account_changeset_grouped = AccountChangeSet::new();
+            let (modules, resources) = account_changeset.into_inner();
+
+            // * Separate the resource groups from the non-resource groups
```

Comment:
Can we document the flow here, including resolving the delta ops, deleting empty resource groups, etc. more prominently as part of finish's documentation?

Reply:
sure